repos/c2sm/packages/flexpart-cosmo/package.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#
+from spack import *
+from distutils.dir_util import copy_tree
+from llnl.util.filesystem import working_dir
+
+
+class FlexpartCosmo(MakefilePackage):
+    """flexpart is a Lagrangian dispersion model"""
+
+    homepage = 'https://github.com/C2SM-RCM/flexpart'
+    git = 'ssh://git@github.com/C2SM-RCM/flexpart.git'
+
+    version('main', branch='testing')
+
+    depends_on('eccodes +fortran')
+    depends_on('netcdf-fortran')
+
+    depends_on('libtool@2.4.6', type='build')
+    depends_on('eccodes@2.25:', type=('build', 'link', 'run'))
+    depends_on('gcc@11:', type=('build', 'run'))
+
+    conflicts('%nvhpc')
+    conflicts('%pgi')
+
+    build_directory = 'src'
+
+    makefile_file = "Makefile.spack"
```

Comment:
Introducing a special makefile for spack would be weird. So maybe it's a left over from testing. Is it?

Reply:
The general [Makefile](https://github.com/C2SM-RCM/flexpart/blob/testing/src/Makefile) was the reason why `spack install` was not working correctly, as it contains many machine-specific setups and probably some variables were not set correctly (and thus causing a different behavior). The [Makefile.spack](https://github.com/C2SM-RCM/flexpart/blob/testing/src/Makefile.spack) contains none of such machine-specific blocks anymore and will stay in the code.

Reply:
Note that other Makefiles will still be used to build Flexpart without spack on specific machines (ddm*: Empa, tsa: MeteoSwiss).